### PR TITLE
Fix interviewer controller and refactoring it

### PR DESCRIPTION
**Pull request by sashav389**

### CODE REVIEW

**Comment by GrEFeRFeeD** [comment]
|File|Coverage [34.07%]|:green_apple:|
|:-|:-:|:-:|
|[InterviewerSlotDtoValidator.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/fc72f6a949c999bf163dc306a543155a4e5eceec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2Finterviewerslot%2FInterviewerSlotDtoValidator.java)|40.14%|:green_apple:|
|[InterviewerController.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/fc72f6a949c999bf163dc306a543155a4e5eceec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FInterviewerController.java)|14.12%|:green_apple:|

|Total Project Coverage|53.11%|:green_apple:|
|:-|:-:|:-:|

**Review by sonnka — APPROVED** [review]

**Comment by GrEFeRFeeD** [comment]
|File|Coverage [34.07%]|:green_apple:|
|:-|:-:|:-:|
|[InterviewerSlotDtoValidator.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/107026fef1d5ca2bd60d1a8fa1278747fefdd892/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2Finterviewerslot%2FInterviewerSlotDtoValidator.java)|40.14%|:green_apple:|
|[InterviewerController.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/107026fef1d5ca2bd60d1a8fa1278747fefdd892/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FInterviewerController.java)|14.12%|:green_apple:|

|Total Project Coverage|54.21%|:green_apple:|
|:-|:-:|:-:|

**Review by GrEFeRFeeD — APPROVED** [review]
Good work.

**Comment by GrEFeRFeeD** [comment]
|File|Coverage [42.17%]|:green_apple:|
|:-|:-:|:-:|
|[PeriodValidator.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/1e43c45a2450c5072b5cd5f82cee7fcfac4ca7aa/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2Fperiod%2Fservices%2Fvalidation%2FPeriodValidator.java)|100%|:green_apple:|
|[InterviewerSlotDtoValidator.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/1e43c45a2450c5072b5cd5f82cee7fcfac4ca7aa/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2Finterviewerslot%2FInterviewerSlotDtoValidator.java)|40.14%|:green_apple:|
|[InterviewerController.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/1e43c45a2450c5072b5cd5f82cee7fcfac4ca7aa/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FInterviewerController.java)|14.12%|:green_apple:|

|Total Project Coverage|54.21%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/controllers/InterviewerController.java [hunk 1]
```diff
@@ -55,32 +54,25 @@ public InterviewerController(
...
-        .interviewerSlotValidateDto(interviewerSlotDto);
-
-    interviewerSlot.getWeek().addInterviewerSlot(interviewerSlot);
-
-    interviewerSlotService.create(interviewerSlot);
-
-    interviewerSlotDto.setInterviewerSlotId(interviewerSlot.getId());
-
+    interviewerSlotDtoValidator
+        .interviewerSlotValidateDtoAndCreate(interviewerSlotDto, authentication, interviewerId);
```
Comment (makskostyshen) [inline src/main/java/com/intellias/intellistart/interviewplanning/controllers/InterviewerController.java:75]: I'd rename method to just validateAndCreate

File: src/main/java/com/intellias/intellistart/interviewplanning/controllers/InterviewerController.java [hunk 2]
```diff
@@ -89,57 +81,43 @@ public ResponseEntity<InterviewerSlotDto> createInterviewerSlot(
...
-        .interviewerSlotValidateDto(interviewerSlotDto);
-    interviewerSlotNew.setId(id);
-
-    interviewerSlotService.create(interviewerSlotNew);
-
-    interviewerSlotDto.setInterviewerSlotId(interviewerSlotNew.getId());
-
-    interviewerSlotNew.getWeek().addInterviewerSlot(interviewerSlotNew);
+    interviewerSlotDtoValidator
+        .interviewerSlotValidateDtoAndUpdate(interviewerSlotDto, authentication, interviewerId,
```
Comment (makskostyshen) [inline src/main/java/com/intellias/intellistart/interviewplanning/controllers/InterviewerController.java:105]: I'd rename method to just validateAndUpdate

**Review by makskostyshen — CHANGES_REQUESTED** [review]
Generally okay. Small method names questions

**Comment by GrEFeRFeeD** [comment]
|File|Coverage [42.17%]|:green_apple:|
|:-|:-:|:-:|
|[PeriodValidator.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/56480a0b911a96bb6177d1fec16ae0c9c0fbdbda/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2Fperiod%2Fservices%2Fvalidation%2FPeriodValidator.java)|100%|:green_apple:|
|[InterviewerSlotDtoValidator.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/56480a0b911a96bb6177d1fec16ae0c9c0fbdbda/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2Finterviewerslot%2FInterviewerSlotDtoValidator.java)|40.14%|:green_apple:|
|[InterviewerController.java](https://github.com/GrEFeRFeeD/intellistart-java-2022-oneweek/blob/56480a0b911a96bb6177d1fec16ae0c9c0fbdbda/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FInterviewerController.java)|14.12%|:green_apple:|

|Total Project Coverage|54.21%|:green_apple:|
|:-|:-:|:-:|

**Review by makskostyshen — APPROVED** [review]
Okay)

**Pull request merged**